lib/js/src/compose/types/page-block/record-list.ts

```diff
@@ -50,25 +50,26 @@ interface Options {
   positionField?: string;
   refField?: string;
   editFields?: unknown[];
-
+  
   // When adding a new record, link it to parent when available
   linkToParent: boolean;
-
+  
   // Should records be opened in a new tab
   // legacy field that has been removed but we keep it for backwards compatibility
   openInNewTab: boolean;
-
+  
   // Are table rows selectable
   selectable: boolean;
   selectMode: 'multi' | 'single' | 'range';
-
+  
   // Ordered list of buttons to display in the block
   selectionButtons: Array<Button>;
-
+  
   bulkRecordEditEnabled: boolean;
   inlineRecordEditEnabled: boolean;
   filterPresets: FilterPreset[];
   showRecordPerPageOption: boolean;
+  openInEditMode: boolean;
```

Comment:
you need to rename this, since based on the name it doesn't really say when this option is applied.
It changes in what mode records are opened. similar to recordDisplayOption it can be called recordDefaultMode or something like that